src/drivers/intel/dmic/dmic_nhlt.c

```diff
@@ -0,0 +1,418 @@
+	} else {
+		/* Clear TIE, SIP, FCI, set FINIT, the rest of bits as such */
+		val = (out_control[1] &
+			~(OUTCONTROL1_TIE_BIT | OUTCONTROL1_SIP_BIT | OUTCONTROL1_FCI_BIT)) |
+			OUTCONTROL1_FINIT_BIT;
+		dai_write(dai, OUTCONTROL1, val);
+		dai_info(dai, "dmic_set_config_nhlt(): OUTCONTROL1 = %08x", val);
+	}
+
+	/* Get PDMx registers */
+	pdm_ctrl_mask = ((struct nhlt_pdm_ctrl_mask *)p)->pdm_ctrl_mask;
+	num_pdm = count_bits_set(pdm_ctrl_mask);
+	p += sizeof(struct nhlt_pdm_ctrl_mask);
+	dai_dbg(dai, "dmic_set_config_nhlt(): pdm_ctrl_mask = %d", pdm_ctrl_mask);
+	if (num_pdm < 1 || num_pdm > DMIC_HW_CONTROLLERS) {
+		dai_err(dai, "dmic_set_config_nhlt(): illegal number of PDMs %d", num_pdm);
+		return -EINVAL;
+	}
+
+	for (n = 0; n < num_pdm; n++) {
+		dai_info(dai, "dmic_set_config_nhlt(): PDM%d", n);
+
+		/* Get CIC configuration */
+		pdm_cfg[n] = (struct nhlt_pdm_ctrl_cfg *)p;
+		p += sizeof(struct nhlt_pdm_ctrl_cfg);
+
+		comb_count = CIC_CONFIG_COMB_COUNT_GET(pdm_cfg[n]->cic_config);
+		p_mcic = comb_count + 1;
+		clkdiv = MIC_CONTROL_PDM_CLKDIV_GET(pdm_cfg[n]->mic_control);
+		p_clkdiv = clkdiv + 2;
+		if (dmic->global->active_fifos_mask == 0) {
+			val = pdm_cfg[0]->cic_control;
+			bf1 = CIC_CONTROL_SOFT_RESET_GET(val);
+			bf2 = CIC_CONTROL_CIC_START_B_GET(val);
+			bf3 = CIC_CONTROL_CIC_START_A_GET(val);
+			bf4 = CIC_CONTROL_MIC_B_POLARITY_GET(val);
+			bf5 = CIC_CONTROL_MIC_A_POLARITY_GET(val);
+			bf6 = CIC_CONTROL_MIC_MUTE_GET(val);
+			bf7 = CIC_CONTROL_STEREO_MODE_GET(val);
+			dai_dbg(dai, "dmic_set_config_nhlt(): CIC_CONTROL = %08x", val);
+			dai_dbg(dai, "  soft_reset=%d, cic_start_b=%d, cic_start_a=%d",
+				bf1, bf2, bf3);
+			dai_dbg(dai, "  mic_b_polarity=%d, mic_a_polarity=%d, mic_mute=%d, stereo_mode=%d",
+				bf4, bf5, bf6, bf7);
+
+			/* Clear CIC_START_A and CIC_START_B, set SOF_RESET and MIC_MUTE*/
+			val = (val & ~(CIC_CONTROL_CIC_START_A_BIT | CIC_CONTROL_CIC_START_A_BIT)) |
+				CIC_CONTROL_SOFT_RESET_BIT | CIC_CONTROL_MIC_MUTE_BIT;
+			dai_write(dai, base[n] + CIC_CONTROL, val);
+			dai_dbg(dai, "dmic_set_config_nhlt(): CIC_CONTROL = %08x", val);
+
+			val = pdm_cfg[n]->cic_config;
+			bf1 = CIC_CONFIG_CIC_SHIFT_GET(val);
+			dai_dbg(dai, "dmic_set_config_nhlt(): CIC_CONFIG = %08x", val);
+			dai_dbg(dai, "  cic_shift=%d, comb_count=%d", bf1, comb_count);
+
+			/* Use CIC_CONFIG as such */
+			dai_write(dai, base[n] + CIC_CONFIG, val);
+			dai_dbg(dai, "dmic_set_config_nhlt(): CIC_CONFIG = %08x", val);
+
+			val = pdm_cfg[n]->mic_control;
+			bf1 = MIC_CONTROL_PDM_SKEW_GET(val);
+			bf2 = MIC_CONTROL_PDM_CLK_EDGE_GET(val);
+			bf3 = MIC_CONTROL_PDM_EN_B_GET(val);
+			bf4 = MIC_CONTROL_PDM_EN_A_GET(val);
+			dai_dbg(dai, "dmic_set_config_nhlt(): MIC_CONTROL = %08x", val);
+			dai_dbg(dai, "  clkdiv=%d, skew=%d, clk_edge=%d", clkdiv, bf1, bf2);
+			dai_dbg(dai, "  en_b=%d, en_a=%d", bf3, bf4);
+
+			/* Clear PDM_EN_A and PDM_EN_B */
+			val = val & ~(MIC_CONTROL_PDM_EN_A_BIT | MIC_CONTROL_PDM_EN_B_BIT);
```

Comment:
`val &= ...`

Reply:
Done, surprisingly this was the only one.